### PR TITLE
engine: use Pod info from the EngineState instead of a separate DeployInfo store

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc,

Please review the following commits I made in branch nicks/ch774/containerupdate:

ba01377e368856bba17ce61327e28bdac0eac5f5 (2018-11-20 17:48:09 -0500)
engine: use Pod info from the EngineState instead of a separate DeployInfo store
This deletes a lot of duplicate code paths that fetch the same data.

As a nice side effect, this fixes the bug where we would try to container-update
only one container if there's more than one pod

d48fb81ab1112db2625b6ef0aa95e9d21d4283cb (2018-11-20 16:06:10 -0500)
engine: Support multiple pods for the same manifest
There are a variety of reasons why you might have multiple pods:
a Deployment with more than 2 replicas, or 2 Deployments using the same
image. It's not super important why they're there, just that we handle
them in a reasonable way.

### CODE REVIEW

**Comment by nicks** [comment]
post-vacation ping!

**Comment by maiamcc** [comment]
Yes, thank you! On it :)

File: internal/engine/synclet_build_and_deployer.go [hunk 1]
```diff
@@ -129,18 +127,22 @@ func (sbd *SyncletBuildAndDeployer) updateViaSynclet(ctx context.Context,
 		return store.BuildResult{}, err
 	}
 
-	sCli, err := sbd.sm.ClientForPod(ctx, deployInfo.podID, state.LastResult.Namespace)
+	sbd.mu.Lock()
+	sbd.deploys[state.LastResult.Image.String()] = deployInfo
+	sbd.mu.Unlock()
+
+	sCli, err := sbd.sm.ClientForPod(ctx, deployInfo.PodID, state.LastResult.Namespace)
```
Comment (maiamcc) [inline internal/engine/synclet_build_and_deployer.go]: put a TODO about warming synclet connections?

File: internal/engine/synclet_build_and_deployer.go [hunk 2]
```diff
@@ -115,12 +117,8 @@ func (sbd *SyncletBuildAndDeployer) updateViaSynclet(ctx context.Context,
...
 	containerPathsToRm := build.PathMappingsToContainerPaths(toRemove)
 
-	deployInfo, err := sbd.dd.DeployInfoForImageBlocking(ctx, state.LastResult.Image)
-
-	// We theoretically already checked this condition :(
-	if err != nil {
-		return store.BuildResult{}, errors.Wrap(err, "deploy info fetch failed")
-	} else if deployInfo.Empty() {
+	deployInfo := state.DeployInfo
+	if deployInfo.Empty() {
```
[the remark this answers is not in the document]
Comment (nicks) [reply internal/engine/synclet_build_and_deployer.go]: eh, this variable gets used several times below, so it's probably simpler as-is

File: internal/engine/synclet_build_and_deployer.go [hunk 3]
```diff
@@ -75,10 +79,8 @@ func (sbd *SyncletBuildAndDeployer) canSyncletBuild(ctx context.Context,
 	}
 
 	// Can't do container update if we don't know what container manifest is running in.
-	info, err := sbd.dd.DeployInfoForImageBlocking(ctx, state.LastResult.Image)
-	if err != nil {
-		return errors.Wrap(err, "deploy info fetch failed")
-	} else if info.Empty() {
+	info := state.DeployInfo
```
[the remark this answers is not in the document]
Comment (nicks) [reply internal/engine/synclet_build_and_deployer.go]: done!

File: internal/engine/synclet_build_and_deployer.go (hunk 1 above)
Comment (nicks) [reply internal/engine/synclet_build_and_deployer.go]: done!

**Pull request merged**